pegasus/sites.v3/code.org/public/educate/csc.md.partial

```diff
@@ -7,28 +7,27 @@ style_min: true
 
 # Heading
 
-First part of content.
+Some description goes here.
 
-## Section 0
+{{ tutorial_promo2, heading: "New Hour of Code activities", background_image: "/images/fit-1940/csc/landing/tutorial_promo2_background.jpg", left_heading: "Mood study", left_description: "Learn about computer science and poetry by creating an animated poem.", left_specs: "", left_button_link: "/moodstudy", left_button_text: "Start", right_heading: "Hello World", right_description: "Learn computer science fundamentals with Sprite Lab", right_specs: "", right_button_link: "/helloworld", right_button_text: "Start" }}
```

Comment:
Do note that if we add this file to the translation pipeline, translators will see the line exactly as presented, and will have free reign to edit it as they please. This includes the `background_image` and the `left_` and `right_button_link` parameters, which could leave us vulnerable to malicious translations.

I don't believe this file is currently in the translation pipeline and I have no idea if we have plans to add it, but I want to call out the risk here. We added the `.partial` syntax specifically to provide a way for us to create content that uses templates but does so in a way that's safe to expose to translators, our thinking being that if we aren't concerned about translator safety then we have the option of using ERB and that if we can't use ERB, a carefully restricted syntax could be a safe way for us to still include ERB subtemplates.

I hadn't considered the extent to which adding support for passing parameters to the partial syntax opens us up to that kind of vulnerability, and now that I have I'm reconsidering whether it was okay for us to have done so in the first place. Can you clarify why you want to use the partial syntax for template, rather than ERB?

Reply:
As discussed, I wanted a really easy way for content editors to be able to edit a single markdown file, via Dropbox, with their user-facing strings being passed into UI components (that are implemented as views/partials, but which content editors wouldn't have to touch).  And all of that markdown and those strings would go through the translation pipeline, i.e. would be localizable.

Reply:
Yeah, that makes sense. I think I hadn't sufficiently thought through this before signing off on the "add arguments" plan. We've fallen into this trap before, where our desire to empower our internal content editors turns into a footgun when we end up giving external translators access to those same tools, however obliquely.

Unfortunately, I'm going to recommend we roll back the change to allow parameters to be passed to the partials syntax and figure out a different way to go about achieving your goals. My apologies for not recognizing this before you put in a bunch of work!

Reply:
@Hamms are malicious translators enough of a concern to remove this functionality that would be so beneficial to content writers? My assumption is the contracted translators we work with could have instructions to not edit/ignore translating sections. I think this is worth discussing further and I will look to schedule a quick sync.

Reply:
Yes, 100%. We do occasionally work with contracted translators, but the translation service we use allows anyone to make an account and submit translations. We essentially allow anyone to make anonymous changes to our content without oversight.

I will add that this partials system was specifically created because we had a need for a way to include templates which was entirely secure against the possibility of malicious use. I recognize that content editors would like the many, many tools currently available to them to be more convenient, but taking the _only solution we have_ in this space for translations and making it unsuitable for that purpose is not an acceptable compromise.

Reply:
A quick note that we discussed and resolved these issues in https://github.com/code-dot-org/code-dot-org/pull/42688.